src/utils/tensor.js

```diff
@@ -198,6 +198,29 @@ export class Tensor {
         return this;
     }
 
+    /**
+     * Return a new Tensor with a callback function applied to each element.
+     * @param {Function} callback - The function to apply to each element. It should take three arguments:
+     *                              the current element, its index, and the tensor's data array.
+     * @returns {Tensor} A new Tensor with the callback function applied to each element.
+     */
+    map(callback) {
+        return this.clone().map_(callback);
+    }
+
+    /**
+     * Apply a callback function to each element of the tensor in place.
+     * @param {Function} callback - The function to apply to each element. It should take three arguments:
+     *                              the current element, its index, and the tensor's data array.
+     * @returns {Tensor} Returns `this`.
+     */
+    map_(callback) {
+        for (let i = 0; i < this.data.length; ++i) {
+            this.data[i] = callback(this.data[i], i, this.data);
+        }
```

Comment:
```suggestion
        const this_data = this.data;
        for (let i = 0; i < this_data.length; ++i) {
            this_data[i] = callback(this_data[i], i, this_data);
        }
```

Reply:
Thanks for the optimisation!  
JavaScript's silent pointers; I never think about this :smile: